rali/rali/source/tf_meta_data_reader_detection.cpp

```diff
@@ -0,0 +1,279 @@
+    std::string user_label_key, std::string user_text_key, 
+    std::string user_xmin_key, std::string user_ymin_key, std::string user_xmax_key, std::string user_ymax_key,
+    std::string user_filename_key)
+{
+    // std::cerr << "The user_label_key is " << user_label_key << ", and the user_filename_key is " << user_filename_key << "\n";
+    // std::cerr << "user_xmin_key is " << user_xmin_key << ", user_ymin_key is " << user_ymin_key << "\n";
+    // std::cerr << "user_xmax_key is " << user_xmax_key << ", user_ymax_key is " << user_ymax_key << "\n";
+    uint length;
+    length = file_contents.tellg();
+
+    std::string temp;
+    size_t uint64_size, uint32_size;
+    uint64_t data_length;
+    uint32_t length_crc, data_crc;
+    uint64_size = sizeof(uint64_t); 
+    uint32_size = sizeof(uint32_t); 
+    char * header_length = new char [uint64_size];
+    char * header_crc = new char [uint32_size];
+    char * footer_crc = new char [uint32_size];
+    file_contents.read(header_length, uint64_size);
+    file_contents.read(header_crc, uint32_size);
+    memcpy(&data_length, header_length, sizeof(data_length));
+    memcpy(&length_crc, header_crc, sizeof(length_crc));
+
+    if(length + data_length + 16 == file_size){
+        _last_rec = true;
+    }
+    char *data = new char[data_length];
+    file_contents.read(data,data_length);
+    tensorflow::Example single_example;
+    single_example.ParseFromArray(data,data_length);
+    tensorflow::Features features = single_example.features();
+
+
+    auto feature = features.feature();
+    tensorflow::Feature single_feature,sf_xmin,sf_ymin,sf_xmax,sf_ymax,sf_fname,sf_label;
+    
+    single_feature = feature.at(user_filename_key);
+    std::string fname = single_feature.bytes_list().value()[0];
+    
+    float bbox_xmin,bbox_ymin,size_b_xmin,bbox_xmax,bbox_ymax;
+    single_feature = feature.at(user_xmin_key);
+    size_b_xmin = single_feature.float_list().value().size();
+    
+    BoundingBoxCords bb_coords;
+    BoundingBoxLabels bb_labels;
+    BoundingBoxCord box;
+
+
+    int label;
+    single_feature = feature.at(user_label_key);
+    label = single_feature.int64_list().value()[0];
+    sf_xmin = feature.at(user_xmin_key);
+    sf_ymin = feature.at(user_ymin_key);
+    sf_xmax = feature.at(user_xmax_key);
+    sf_ymax = feature.at(user_ymax_key);
+    for(int i=0;i<size_b_xmin;i++)
+    {
+      bbox_xmin = sf_xmin.float_list().value()[i];
+      bbox_ymin = sf_ymin.float_list().value()[i];
+      bbox_xmax = sf_xmax.float_list().value()[i];
+      bbox_ymax = sf_ymax.float_list().value()[i]; 
+      box.x = bbox_xmin;
+      box.y = bbox_ymin;
+      box.w = bbox_xmax;
+      box.h = bbox_ymax;
+      bb_coords.push_back(box);
+      bb_labels.push_back(label);
+      add(fname, bb_coords, bb_labels);
+      bb_coords.clear();
+      bb_labels.clear();
+    }
+
+    
+
+    
+    
+    file_contents.read(footer_crc, sizeof(data_crc));
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)